prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -181,64 +185,74 @@ class Hl7Serializer(val metadata: Metadata) {
         hapiContext.modelClassFactory = mcf
         val parser = hapiContext.pipeParser
         val reg = "(\r|\n)".toRegex()
-        val cleanedMessage = reg.replace(message, "\r")
-        val hapiMsg = parser.parse(cleanedMessage)
-        val terser = Terser(hapiMsg)
-        schema.elements.forEach {
-            if (!mappedRows.containsKey(it.name))
-                mappedRows[it.name] = mutableSetOf()
-
-            if (it.hl7Field.isNullOrEmpty() && it.hl7OutputFields.isNullOrEmpty()) {
-                mappedRows[it.name]?.add("")
-                return@forEach
-            }
+        val cleanedMessage = reg.replace(message, "\r").trim()
+        // if the message is empty, return a row result that warns of empty data
+        if (cleanedMessage.isEmpty()) {
+            return RowResult(emptyMap(), emptyList(), listOf("Cannot parse empty message"))
+        }
 
```

Comment:
Add a check in the parsing step to return an empty row result before trying to invoke the parser.